Add SplitEVDAQ

MorganAskins opened the pull request:
New simulated DAQ which creates trigger pulses for each pmt hit of a
    fixed width (pulse_width).  After a pmt is hit, it cannot produce a new
    pulse for a specified time (pmt_lockout). When the sum of all concurrent
    pulses goes above threshold (trigger_threshold), the daq produces a
    global trigger to record all hits within the trigger window
    (trigger_window).  Afterwards the daq cannot fire another trigger until
    (trigger_lockout) time after the end of the trigger window. The
    precision of the trigger is set by the trigger_resolution.

    This DAQ can handle any number of coincidence events produced by a
    generator (such as the two signals from inverse beta decay), and will
    also properly retrigger on late light depending on the trigger window
    and lockout settings.

MorganAskins commented:
Not ready to merge. Currently working on this in public rat-pac, will merge here only after it's merged and working as intended there.

MarcFBergevin commented on src/daq/src/SplitEVDAQProc.cc:
```diff
@@ -0,0 +1,161 @@
+  vector<double> triggerTimes;
+  for(int i=0; i<nbins; i++)
+  {
+    double v = triggerHistogram[i];
```
double v should be defined outside of loop

MorganAskins replied:
v is the histogram entry being looped over in this case, so should be okay inside.

MorganAskins commented:
Changed the recorded PMT time to be relative to the trigger time and not the clock. This fixes the bonsai reconstruction for neutrons in my tests.

MorganAskins commented:
In getting ready for noise hits; added the functionality to skip noise hits when building triggers (if desired) -- this mimics the old system of trigger events without noise but still includes the noise in reconstruction. Also added the macro commands via the SetI and SetD functions.

MarcFBergevin commented:
There is currently no flag to distinguish veto from inner PMTs. This will be less of an issue in the near future. 

Also, could you add:
`#/rat/db/set SplitEVDAQ pulse_width 100
#/rat/db/set SplitEVDAQ trigger_threshold 4.0
#/rat/db/set SplitEVDAQ trigger_window 400.0
#/rat/db/set SplitEVDAQ pmt_lockout 400.0
#/rat/db/set SplitEVDAQ trigger_lockout 0.0
#/rat/db/set SplitEVDAQ trigger_resolution 1.0
#/rat/db/set SplitEVDAQ lookback 400.0
#/rat/db/set SplitEVDAQ max_hit_time 1000000.0
#/rat/db/set SplitEVDAQ pmt_type 1
#/rat/db/set SplitEVDAQ trigger_on_noise 1

/run/initialize`

to the official macros? Where the desired option to change can simply be uncommented out.
